substrate-archive-backend/src/runtime_version_cache.rs

```diff
@@ -159,9 +160,11 @@ impl<B: BlockT> VersionRange<B> {
 }
 
 fn decode_version(version: &[u8]) -> Result<sp_version::RuntimeVersion> {
+	#[allow(clippy::redundant_slicing)]
```

Comment:
I've tried a few variations of `::decode(&mut &version[..])` and the one clippy accepts is to pass in a `mut version`, ultimately decoding `&mut version` (without [..]).

However, the version clippy accepts leads to a runtime error where we can no longer decode the RuntimeVersion. I believe it has to do with this: https://github.com/rust-lang/rust-clippy/issues/6968

Reply:
You can just reborrow the slice as that's what clippy will be suggesting after it's fixed. i.e. `&mut &*version`